server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -801,6 +802,16 @@ public void handleException(TransportException e) {
                 return;
             }
 
+            if (cause instanceof CircuitBreakingException) {
+                retryRecovery(
+                    recoveryId,
+                    "remote shard responded with circuit_breaking_exception",
+                    recoverySettings.retryDelayStateSync(),
```

Comment:
`retryDelayStateSync()` (defaults to 500ms) seems too short, we should wait much longer here. `retryDelayNetwork()` (defaults to 5s) seems better, but not exactly appropriate either. Also I think we'd like a log (INFO? WARN?) in this case

Reply:
I missed that `retryDelayStateSync()` is so short. I agree that doesn't make much sense. I am not sure what would be a good value here, but the 5s seems reasonable to reuse. Not sure if it worths its own setting, and also probably better not to hardcode it. There is a log, but not in the right level. I agree WARN is probably more fitting.